web-console/src/views/query-view/query-view.tsx

```diff
@@ -320,14 +312,15 @@ export class QueryView extends React.PureComponent<QueryViewProps, QueryViewStat
       processQuery: async (queryWithContext: QueryWithContext) => {
         const { queryString, queryContext, wrapQueryLimit } = queryWithContext;
 
-        const actualQuery = QueryView.wrapInLimitIfNeeded(queryString, wrapQueryLimit);
-
         const explainPayload: Record<string, any> = {
-          query: QueryView.wrapInExplainIfNeeded(actualQuery),
+          query: QueryView.wrapInExplainIfNeeded(queryString),
           resultFormat: 'object',
         };
 
-        if (!isEmptyContext(queryContext)) explainPayload.context = queryContext;
+        if (!isEmptyContext(queryContext) || wrapQueryLimit) {
+          explainPayload.context = queryContext || {};
```

Comment:
```suggestion
          explainPayload.context = Object.assign({}, queryContext || {});
```

Otherwise you are changing the queryContext state variable in place if it is set. Maybe not but still cleaner to avoid even the possibility of that.

Reply:
changed